### PR TITLE
Added the authors behaviors

### PR DESCRIPTION
**Pull request by ale-rt**

### CODE REVIEW

**Comment by ale-rt** [comment]
QA instructions:

1. reinstall the package (no upgrade steps yet)
2. Create some persone in the site root
3. Create a Publication in the site root
4. Crete a Volume in the Publication
5. Three different "Review" types can be added: they should all have an authors field that can pick the previously added persons

**Comment by ale-rt** [comment]
The changes to the add permissions, i.e.:
```
<property name="add_permission">cmf.AddPortalContent</property>
```
are due to the fact that we do not have the mentioned permissions.

Without these changes the custom portal types are not addable.

In the old system there were custom permission, but it seems to me that the [rolemap](https://github.com/syslabcom/recensio.contenttypes/blob/acf6ca3c962bfcf50600739087973de3ba7ad124/recensio/contenttypes/profiles/default/rolemap.xml) suggests to use the standard one, which is already something that a Contributor has.

File: src/recensio/plone/behaviors/configure.zcml [hunk 1]
```diff
@@ -3,4 +3,12 @@
     xmlns:plone="http://namespaces.plone.org/plone"
     >
 
+  <plone:behavior
+      name="recensio.authors"
+      title="Authors Behavior"
+      factory=".authors.Authors"
+      provides=".authors.IAuthors"
+      for="plone.dexterity.interfaces.IDexterityContent"
```
Comment (thet) [inline src/recensio/plone/behaviors/configure.zcml:11]:
```suggestion
```

File: src/recensio/plone/behaviors/configure.zcml [hunk 2]
```diff
@@ -3,4 +3,12 @@
     xmlns:plone="http://namespaces.plone.org/plone"
     >
 
+  <plone:behavior
+      name="recensio.authors"
+      title="Authors Behavior"
+      factory=".authors.Authors"
```
Comment (thet) [inline src/recensio/plone/behaviors/configure.zcml:9]:
```suggestion
```
I think we do not need a factory for this behavior, right?

File: src/recensio/plone/behaviors/authors.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    def __init__(self, context):
+        self.context = context
+
+    @property
+    def authors(self):
+        return self.context.authors
+
+    @authors.setter
+    def authors(self, value):
+        self.context.authors = value
```
Comment (thet) [inline src/recensio/plone/behaviors/authors.py:43]:
```suggestion
```
I think we do not need a factory for this behavior, right?

**Review by thet — CHANGES_REQUESTED** [review]
I‌ think we can leave out the behavior factory unless you plan to extend it in the future.

Otherwise LGTM

File: src/recensio/plone/behaviors/configure.zcml (hunk 1 above)
Comment (ale-rt) [reply src/recensio/plone/behaviors/configure.zcml:11]: Plone does like I did and I do like Plone: https://github.com/plone/plone.app.dexterity/blob/b46fd64b8a76988d49bee10938eb318f6d302fbd/plone/app/dexterity/behaviors/configure.zcml#L13-L20

File: src/recensio/plone/behaviors/configure.zcml (hunk 2 above)
Comment (ale-rt) [reply src/recensio/plone/behaviors/configure.zcml:9]: Plone does like I did and I do like Plone: https://github.com/plone/plone.app.dexterity/blob/b46fd64b8a76988d49bee10938eb318f6d302fbd/plone/app/dexterity/behaviors/configure.zcml#L13-L20

File: src/recensio/plone/behaviors/authors.py (hunk 3 above)
Comment (ale-rt) [reply src/recensio/plone/behaviors/authors.py:43]: See above

**Comment by thet** [comment]
@ale keeping the behavior factories for no pressing reason is against the "keep things simple" rule. The metadata behavior factories in plone.app.dexterity all do have a purpose. But the behaviors for IAllowDiscussion, IExcludeFromNavigation, etc do not have behavior factories.

If we keep the behavior factories we should be consistent and add those for all.

I'm not dogmatic on this but I‌ prefer not to see more code than necessary, just like you :)

**Comment by thet** [comment]
there is also:
```
marker
    A marker interface to be applied by the behavior. If factory is not given, then this is optional and defaults to the value of provides. If factory is given marker is required and should be different from provides - even if its not enforced.
```
That's actually an argument to use factories really only when needed. Because we ideally would need another marker interface. This is a part in plone.dexterity which I do not like.
Let's just not use factories and we're good - unless we need them.

**Pull request merged**